src/app/features/theme-drawer/theme-drawer.tsx

```diff
@@ -0,0 +1,19 @@
+import { ControlledDrawer } from '@app/components/drawer/controlled-drawer';
+import { useDrawers } from '@app/common/hooks/use-drawers';
+import { useShowThemeSwitcherStore } from '@app/store/ui/ui.hooks';
+import { ThemeList } from './theme-list';
+
+export const ThemesDrawer = () => {
+  const { setIsShowingThemeSwitcher } = useDrawers();
+  const [isShowing] = useShowThemeSwitcherStore();
+
+  return (
+    <ControlledDrawer
+      title="Select Theme"
+      isShowing={isShowing}
+      onClose={() => setIsShowingThemeSwitcher(false)}
+    >
+      <ThemeList />
+    </ControlledDrawer>
+  );
+};
```

Comment:
Makes sense you've done it this way, using controlled drawer like done elsewhere, though imo a better implementation would leverage the router, rather than manually controlling state that points to a location.

Could you have a look at how the _Sign out_ modal is shown. It uses the `BaseDrawer` component and assigns it a route. I hope we can completely remove these `is drawer open` atoms eventually.

Reply:
Good point here, I also refactored the `IncreaseFeeDrawer` to a route so can ref that as well.

Reply:
I think I understood how it works, will change.

Reply:
While reviewing the signout's drawer, I believe I spotted references to `setShowSignOut` that aren't actually doing anything, so I removed them. Seems to work fine without this code.